### ISSUE TITLE
md-not-found not working in 0.10.1

### ISSUE THREAD

**Issue by DaveRevay2** [angular/material#3876]
Demo is first example here:
https://material.angularjs.org/0.10.1-rc3/#/demo/material.components.autocomplete

md-not-found does not show anymore.

**Comment by NaKroTeK** [comment]
Hello, 
same problem here but it work with 0.9.8 : 
https://material.angularjs.org/0.9.8/#/demo/material.components.autocomplete

**Comment by ACDN** [comment]
It is working in the head version of the documents. https://material.angularjs.org/latest/#/demo/material.components.autocomplete

However, it doesn't work in my site after downloading build: version 0.10.1-rc3-master-d270684

**Comment by dmackerman** [comment]
Can confirm. I'm returning an empty array and `<md-not-found>` never displays.

**Comment by ACDN** [comment]
Hey Guys,

Just tried this out again in 0.10.1-RC5 and md-not-found is still not working. 

ACDN

**Comment by sricc** [comment]
Same here..

**Comment by michaelkrog** [comment]
+1

**Comment by emgolubev** [comment]
+1
0.10.1-rc5

**Comment by webley** [comment]
+1

**Comment by wrench7** [comment]
+1

**Comment by ACDN** [comment]
Hey everyone, 

If you really need this piece of functionality working then you are in the same boat as me.  I got fed up waiting for this bug to get fixed and took it upon myself to find the cause.  It turned out to be a logic error.  One that I am sure we have all made a million times.

Below is the offending code. 

```javascript

  /**
   * Determines if the menu should be hidden.
   * @returns {boolean}
   */
  function shouldHide () {
    if (!isMinLengthMet() || !ctrl.matches.length) return true;
  }

```

It's a function that determines if the auto complete drop down should be hidden. The problem is the OR operator in the if expression.  The logic doesn't work. For example, if I type 1 character into the auto complete and that character is not in the data source (no matches) the expression looks like this:  if (!1 || !0)  This ends up being true, instead of false and the drop down is hiddedn.  The no results template is shown as a list item so we need to alter this function in order to get the results we want. 

Below is the modified version.  

```javascript

  /**
   * Determines if the menu should be hidden.
   * @returns {boolean}
   */
  function shouldHide () {
    if (!isMinLengthMet() && !ctrl.matches.length) return true;
  }

```

Changing the OR to an AND operator causes the logic to start working and return FALSE in our example.  if (!1 && !0) returns false.  Which instructs the MdAutoCompleteCtrl to show the menu.

I've tested this pretty thoroughly with my use cases an it is working great.  I won't claim that this fix will work for everyone but give it a shot a see what happens.

Lastly, a word to the material design team. First off you guys are doing amazing work and my project would be nothing without this project.  However, I am concerned about the priorities in this project.  I have to imagine there are lots of projects in dev or live that depend on this library. While new features are essential I would say keeping existing functionality working and resolve bugs on previously released components is critical.

**Comment by WieldSoftwareDevelopement** [comment]
Hey guys,

Glad to see there were progress on this issue in release: release: version 0.10.1-master-97a8f6d

The md-not-found directive now shows up as expected!

Though there is a new issue. Which I'll log as a separate issue.

**Comment by mohitadwani** [comment]
+1
This is fixed in v0.10.1-master-e26a275
But now when it shows no results found, the dropdown doesn't close when you go out of focus
